cli/lib/kontena/cli/stacks/upgrade_command.rb

```diff
@@ -198,6 +198,7 @@ def run_installs(changes)
         caret "Installing new dependency #{cmd.last} as #{added_stack}"
         deployable_stacks << added_stack
         Kontena.run!(cmd)
+        set_env_variables(stack_name, current_grid)
```

Comment:
This shouldn't strictly be necessary, I think? Yes, the `ENV['stack']` will leak from the `Kontena.run(['kontena', ',stack', 'install', ..])`, but OTOH all of the stacks reader executes happen before that. After the `run_installs`, I don't think the `run_upgrades`, `run_deploys` and `run_removes` would care about that `ENV['stack']` anymore?